### PR TITLE
Utilized redux

### PR DESCRIPTION
**Pull request by thankgodr**
This PR includes 

- The use of the useSelector hook
- the use of the useDispatch hook
- consume action created by redux
- Code linter
- Add book Functionality
- Remove Book Functionality
- Update status functionality

### CODE REVIEW

File: src/pages/Books.js [hunk 1]
```diff
@@ -1,47 +1,25 @@
...
-  };
-
-  const updateBook = (book) => {
-    // Todo implement remove book
-    parseInt(book, 10);
-  };
-
+const Books = () => {
+  const books = useSelector((state) => state.books);
+  console.log(books.books);
```
Comment (descholar-ceo) [inline src/pages/Books.js]: - [x] Kindly remove this `console.log`
```suggestion
```

File: src/components/AddNewBook.js [hunk 2]
```diff
@@ -1,14 +1,59 @@
...
+
+            <button
+                type="submit"
+                className="btn btn-primary mb-2"
+                onClick={(e) => {
+                  e.preventDefault();
+                  const tempBookModel = new BookModel(newBook.bookName, newBook.author, 'Testing');
+                  dispatch(bookAction(
+                    { type: ADD_BOOK, book: JSON.stringify(tempBookModel) },
+                  ));
```
Comment (descholar-ceo) [inline src/components/AddNewBook.js]: ### You were expected to have:
- [x] A local state with  title, author where you should have your inputs update the local state every time the inputs are updated (onChange)
- [x] A `submitBookToStore` that takes the inputs from the state and generate a unique `id` and add them to an object, then dispatches the `addBook` action creator (name of the function can be different) with the object as its argument

File: src/components/Book.js [hunk 3]
```diff
@@ -1,53 +1,75 @@
...
+                        <li>
+                            <button onClick={
+                                (event) => {
+                                  event.preventDefault();
+                                  dispatch(bookAction(
+                                    {
+                                      type: REMOVE_BOOK,
+                                      id: props.book.id,
+                                    },
+                                  ));
```
Comment (descholar-ceo) [inline src/components/Book.js]: ### Here also you were expected to have:
- [x] A * `removeBookFromStore` function that takes the id from the book data, and dispatches the `removeBook` action creator (name of the function can be different) with the id as its argument

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @thankgodr 🤝 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
👍🏽 Linters installed and they pass the checks ✔️ 
👍🏽 Professional PR with correct commit message ✔️ 
👍🏽 Descriptive PR comment ✔️ 
👍🏽 Correct git/github flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._
- [ ] You are missing changes related to `book category` on this branch, we will need to see them on this branch so that we can review them as well
- [ ] Kindly fix this section from your readme file in order to have a complete and professional repository
Screen Shot 2022-12-13 at 23 23 34

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @thankgodr, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**